### PR TITLE
Fill the disk before asking for tests

### PR DESCRIPTION
**Pull request by faithanalog**
It's unclear whether we need the `after=` so ive left that commented for now.

should we have `/dev/zero` or `/dev/urandom`? `/dev/zero` takes less guest CPU, and might run faster if crucible can keep up. either way it's encrypted downstairs, so zfs can't optimize out the zeroes. but do we care for any other reason?

### CODE REVIEW

**Comment by ahl** [comment]
Filling with zeros does not seem like a good idea. I'd suggest using fio to write random data from the start to the end of the device. That's what we do for doing performance measurements of new hardware SSDs.

**Comment by faithanalog** [comment]
> Filling with zeros does not seem like a good idea. I'd suggest using fio to write random data from the start to the end of the device. That's what we do for doing performance measurements of new hardware SSDs.

urandom it is then

**Comment by faithanalog** [comment]
is there any particular reason you think using fio is better than dd for filling the disk initially

**Comment by ahl** [comment]
I'm sure either is fine. We're doing this once and it shouldn't take long

File: nixos/jobs.nix [hunk 1]
```diff
@@ -124,6 +124,17 @@
...
+          # fill the disk before running any job, or indeed before asking for
+          # any jobs to run
+
+          # need to wait for the symlink to finish up being added by udev
+          udevadm settle
+
+          # this is expected to fail once dd hits the end of the device, since
+          # we are not specifying a count. It'll fill the whole thing first, and
+          # this way it works no matter the size of it
+          dd if=/dev/urandom of=/dev/cobblestone bs=1M oflag=direct || true
```
Comment (iliana) [inline nixos/jobs.nix:137]: Wait, does this exit non-zero? That's not what I would expect (and I would be worried about masking an error where we didn't finish filling the disk).

**Pull request merged**